td-reproducible-tool/src/main.rs

```diff
@@ -0,0 +1,214 @@
+        // Nothing to check
+    }
+
+    // Solve the path
+    let binary_path: PathBuf = [
+        if workspace == "" { "." } else { &workspace },
+        "target",
+        &target,
+        if profile == "" { "release" } else { &profile },
+        &binary_name,
+    ]
+    .iter()
+    .collect();
+
+    assert!(binary_path.exists());
+    println!("INFO: Found the compiled file: {:?}", binary_path);
+
+    // Load the binary
+    let mut binary =
+        File::open(binary_path.clone()).expect("Failed to open the compiled binary!\n");
+    let mut buf = Vec::new();
+    binary
+        .read_to_end(&mut buf)
+        .expect("Failed to read the compiled binary!\n");
+
+    // Strip time data stamp in PE case
+    if target == "x86_64-unknown-uefi"
+        || binary_name.ends_with("exe")
+        || binary_name.ends_with("efi")
+    {
+        let pe_sig_off =
+            u32::from_le_bytes(buf[PE_SIG_PTR_OFF..PE_SIG_PTR_OFF + 4].try_into().unwrap());
+        let pe_sig = u32::from_le_bytes(
+            buf[pe_sig_off as usize..pe_sig_off as usize + 4]
+                .try_into()
+                .unwrap(),
+        );
+
+        assert_eq!(
+            pe_sig,
+            u32::from_le_bytes([b'P', b'E', b'\0', b'\0']),
+            "Found invalid PE signature!"
+        );
+
+        let time_stamp = u32::from_le_bytes(
+            buf[pe_sig_off as usize + TIMEDATASTAMP_OFF_TO_PE_SIG
+                ..pe_sig_off as usize + TIMEDATASTAMP_OFF_TO_PE_SIG + 4]
+                .try_into()
+                .unwrap(),
+        );
+
+        println!("INFO: Detected TimeDateStamp: {:?}", time_stamp);
+
+        buf[pe_sig_off as usize + TIMEDATASTAMP_OFF_TO_PE_SIG
+            ..pe_sig_off as usize + TIMEDATASTAMP_OFF_TO_PE_SIG + 4]
+            .zeroize();
+
+        let time_stamp = u32::from_le_bytes(
+            buf[pe_sig_off as usize + TIMEDATASTAMP_OFF_TO_PE_SIG
+                ..pe_sig_off as usize + TIMEDATASTAMP_OFF_TO_PE_SIG + 4]
+                .try_into()
+                .unwrap(),
+        );
+        println!("INFO: After removing TimeDateStamp: {:?}", time_stamp);
+    }
+
+    // No more action is needed if strip_path is not specified.
+    if !strip_path {
+        println!(
+            "INFO: -s or --strip_path is not specified, Skipping strip related rust file path."
+        );
+        return Ok(());
+    }
+
+    // Check out CARGO_HOME and RUSTUP_HOME, proceed to strip rust file path.
+    cargo_home = if cargo_home == "" {
+        if env::var("CARGO_HOME").is_err() {
+            panic!("Neither --cargo_home nor system environment for \"CARGO_HOME\" is found!\n")
+        } else {
+            env::var("CARGO_HOME").unwrap()
+        }
+    } else {
+        cargo_home
+    };
+
+    rustup_home = if rustup_home == "" {
+        if env::var("RUSTUP_HOME").is_err() {
+            panic!("Neither --rustup_home nor system environment for \"RUSTUP_HOME\" is found!\n")
```

Comment:
still panic?

Reply:
OK. I did not found -s or --strip_path in readme.

Please add it.

Reply:
> OK. I did not found -s or --strip_path in readme.
> 
> Please add it.

Sorry not to update readme. Now added, please check the latest readme.md